### PR TITLE
feat(core-flows,medusa,types): create/update workflows to create and update PriceList  prices

### PR DESCRIPTION
**Pull request by riqwan**
what:

- create and update workflows for price list prices
- update price list endpoint does a "set"

### CODE REVIEW

File: packages/core-flows/src/price-list/steps/update-price-list-prices.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+      priceListPricesToUpdate.push({
+        price_list_id: id,
+        prices: pricesToUpdate,
+      })
+    }
+
+    const priceIds = priceListPricesToUpdate
+      .map((priceListData) => priceListData.prices.map((price) => price.id))
+      .flat(1)
+      .filter(Boolean)
```
Comment (olivermrbl) [inline packages/core-flows/src/price-list/steps/update-price-list-prices.ts]: suggestion: Move both the mapping and filtering to the loop just before:

```ts
let priceIds: string[] = []
for (const priceListData of data) {
  const pricesToUpdate: UpdatePriceListPriceDTO[] = []
  const { prices = [], id } = priceListData

  for (const price of prices) {
    pricesToUpdate.push({
      ...price,
      price_set_id: variantPriceSetMap[price.variant_id!],
    })

    // This should take care of both no?
    if (price.id) {
      priceIds.push(price.id)
    }
  }

  priceListPricesToUpdate.push({
    price_list_id: id,
    prices: pricesToUpdate,
  })
}
```

**Review by olivermrbl — APPROVED** [review]
LGTM

File: packages/core-flows/src/price-list/steps/update-price-list-prices.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  IPricingModuleService,
+  PriceSetMoneyAmountDTO,
+  UpdatePriceListPriceDTO,
+  UpdatePriceListPricesDTO,
+  UpdatePriceListPricesWorkflowDTO,
+} from "@medusajs/types"
+import { buildPriceSetPricesForModule } from "@medusajs/utils"
+import { StepResponse, createStep } from "@medusajs/workflows-sdk"
+
+type WorkflowStepInput = {
```
Comment (adrien2p) [inline packages/core-flows/src/price-list/steps/update-price-list-prices.ts]: too: rename to something close to the step ?

**Review by adrien2p — APPROVED** [review]
🔥 just few comments to handle

File: packages/core-flows/src/price-list/steps/update-price-list-prices.ts (hunk 1 above)
Comment (riqwan) [reply packages/core-flows/src/price-list/steps/update-price-list-prices.ts]: nice!

File: packages/core-flows/src/price-list/steps/create-price-list-prices.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    if (!createdIds?.length) {
+      return
+    }
+
+    const pricingModule = container.resolve<IPricingModuleService>(
+      ModuleRegistrationName.PRICING
+    )
+
+    if (createdIds.length) {
+      await pricingModule.removePrices(createdIds)
```
Comment (sradevski) [inline packages/core-flows/src/price-list/steps/create-price-list-prices.ts:58]: Would this be `removePriceListPrices`?

File: packages/core-flows/src/price-list/steps/create-price-list-prices.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  CreatePriceListPricesWorkflowStepDTO,
+  IPricingModuleService,
+} from "@medusajs/types"
+import { StepResponse, createStep } from "@medusajs/workflows-sdk"
+
+export const createPriceListPricesStepId = "create-price-list-prices"
+export const createPriceListPricesStep = createStep(
+  createPriceListPricesStepId,
+  async (stepInput: CreatePriceListPricesWorkflowStepDTO, { container }) => {
+    const { data, variant_price_map: variantPriceSetMap } = stepInput
```
Comment (sradevski) [inline packages/core-flows/src/price-list/steps/create-price-list-prices.ts:14]: A general question: Should the diffing of what to create and what to remove be done in the module itself?

File: packages/core-flows/src/price-list/steps/create-price-list-prices.ts (hunk 3 above)
Comment (riqwan) [reply packages/core-flows/src/price-list/steps/create-price-list-prices.ts:58]: We don't have a scoped method for that yet. Both price set and price list prices are the same entity, so its ok to do this for now. Will revisit this when that becomes available.

File: packages/core-flows/src/price-list/steps/create-price-list-prices.ts (hunk 4 above)
Comment (riqwan) [reply packages/core-flows/src/price-list/steps/create-price-list-prices.ts:14]: My personal pref would be for the module to keep things atomic, but I don't have a strong opinion on this. Feel free to put it up on the eng channel and check with the others

**Pull request merged**